### ISSUE TITLE
Feature request: add sniff support for constant comments

### ISSUE THREAD

**Issue by rjgwiz** [squizlabs/PHP_CodeSniffer#2221]
[phpdoc](http://docs.phpdoc.org/glossary.html#term-structural-elements) mentions constants under the programming constructs that should be preceded by a DocBlock.

I can't find anyway to enforce docblock comments for constants.

Docblock comments are currently enforceable via `ClassCommentSniff`, `FunctionCommentSniff`, `VariableCommentSniff`, and `FileCommentSniff`. None seem appropriate to add support for constants (`VariableCommentSniff` would be the closest, but it extends `AbstractVariableSniff` and adding `T_CONST` to the listen list would break a lot of other sniffs, class constants aren't normal variables, and it wouldn't cover defined constants.

Some sniffs exist for constants, for example `UpperCaseConstantNameSniff`. Any hesitations around adding a new sniff called `ConstantCommentSniff`? Scope would be the same as `UpperCaseConstantNameSniff` so any constant defined via the `define()` function or the `const` keyword (e.g. class constant).

With buy-in I could work on `ConstantCommentSniff` and submit a PR when it's ready.

**Comment by gsherwood** [comment]
I think a ConstantComment sniff sounds like a good idea.

I haven't reviewed the doc comment sniffs because I've been waiting for PSR-5 to be complete so I could (hopefully) standardise the checks instead of using the Squiz-specific ones. That, and a lack of time.

If you want to give any of the comment sniffs a go, please do. Huge bonus if you have it enforce PSR-2's recommendations.

**Comment by rjgwiz** [comment]
Thank you @gsherwood. I hope to start working on this soon.

**Comment by TomasVotruba** [comment]
@rjgwiz :+1: You can basically copy https://github.com/Symplify/Symplify/blob/master/packages/CodingStandard/src/Sniffs/Commenting/VarConstantCommentSniff.php + extends a bit to match your needs.

```php
/**
 * @var int
 */
private const VALUE = 10;
```

It works great with static analysis rule - https://github.com/Symplify/Symplify/blob/master/packages/PHPStanExtensions/src/Rules/Classes/MatchingTypeConstantRule.php, e.g.:

```php
/**
 * @var int
 */
private const VALUE = '10'; // Error!
```

**Comment by jrfnl** [comment]
Related to #2222 which will make creating this sniff so much easier ;-)

**Comment by rjgwiz** [comment]
Thank you both. I plan to get to this in the next week or two.

**Comment by TomasVotruba** [comment]
@jrfnl Any ETA on that feature coming to tagged version?

**Comment by jrfnl** [comment]
@TomasVotruba I expect it to be ready for PHPCS 3.5.0 which includes lots of other goodies which will be useful for this too. See #2189

**Comment by TomasVotruba** [comment]
@jrfnl That would be amazing!

**Comment by dereuromark** [comment]
I guess this wasnt pursued further so far?
The links above to the existing ones are 404 by now. Maybe someone knows the new URLs?
